code/parachain/runtime/dali/src/lib.rs

```diff
@@ -1196,6 +1211,157 @@ impl cosmwasm::Config for Runtime {
 	type WeightInfo = cosmwasm::weights::SubstrateWeight<Runtime>;
 }
 
+parameter_types! {
+	pub const RelayChainId: RelayChain = RelayChain::Rococo;
+	pub const SpamProtectionDeposit: Balance = 1_000_000_000_000;
+	pub const MinimumConnectionDelay: u64 = 0; // well we don't need the security tbh.
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Default)]
+pub struct Router {
+	pallet_ibc_ping: pallet_ibc_ping::IbcModule<Runtime>,
+}
+
+impl ModuleRouter for Router {
+	fn get_route_mut(&mut self, module_id: &impl Borrow<ModuleId>) -> Option<&mut dyn Module> {
+		match module_id.borrow().to_string().as_str() {
+			pallet_ibc_ping::MODULE_ID => Some(&mut self.pallet_ibc_ping),
+			_ => None,
+		}
+	}
+
+	fn has_route(module_id: &impl Borrow<ModuleId>) -> bool {
+		matches!(module_id.borrow().to_string().as_str(), pallet_ibc_ping::MODULE_ID)
+	}
+
+	fn lookup_module_by_port(port_id: &PortId) -> Option<ModuleId> {
+		match port_id.as_str() {
+			pallet_ibc_ping::PORT_ID => ModuleId::from_str(pallet_ibc_ping::MODULE_ID).ok(),
+			_ => None,
+		}
+	}
+}
+
+pub struct IbcDenomToAssetIdConversion;
+
+impl DenomToAssetId<Runtime> for IbcDenomToAssetIdConversion {
+	type Error = DispatchError;
+
+	fn from_denom_to_asset_id(denom: &String) -> Result<CurrencyId, Self::Error> {
+		let denom_bytes = denom.as_bytes().to_vec();
+		if let Some(id) = IbcDenoms::<Runtime>::get(&denom_bytes) {
+			return Ok(id)
+		}
+
+		// let pallet_id: AccountId = PalletId(*b"pall-ibc").into_account_truncating();
+
+		// let symbol = denom
+		// 	.split("/")
+		// 	.last()
+		// 	.ok_or_else(|| DispatchError::Other("denom missing a name"))?
+		// 	.as_bytes()
+		// 	.to_vec();
+		// let asset_id = generate_asset_id()?;
+
+		// <assets::Pallet<Runtime> as Create<AccountId>>::create(
+		// 	asset_id,
+		// 	pallet_id.clone(),
+		// 	true,
+		// 	1,
+		// )?;
+		//
+		// <assets::Pallet<Runtime> as Mutate<AccountId>>::set(
+		// 	asset_id,
+		// 	&pallet_id,
+		// 	denom_bytes,
+		// 	symbol,
+		// 	12,
+		// )?;
+		// TODO: the code above is outdated
+		let asset_id =
+			<currency_factory::Pallet<Runtime> as CurrencyFactoryT>::create(RangeId::IBC_ASSETS)?;
+
+		IbcDenoms::<Runtime>::insert(denom_bytes.clone(), asset_id);
+		IbcAssetIds::<Runtime>::insert(asset_id, denom_bytes.clone());
+
+		let location = XcmAssetLocation::new(
+			MultiLocation::new(1, X1(Junction::GeneralIndex(asset_id.into()))).into(),
+		);
+		assets_registry::Pallet::<Runtime>::set_reserve_location(
+			asset_id,
+			location,
+			Rational64::one(),
+			Some(12),
+		)?;
```

Comment:
I think this will work for now; but @PoisonPhang is working on a proposal to change currency-factory and such.